pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1440,6 +1440,17 @@ private void fetchLatestStreamOffset() {
     }
   }
 
+  public StreamPartitionMsgOffset fetchLatestStreamOffset(Long maxWaitTimeMs) {
```

Comment:
Seems we are always using 5s as the wait time. So we should either:
1. Make `fetchLatestStreamOffset()` public and use it
2. Remove `fetchLatestStreamOffset()`, and add a `DEFAULT_MAX_WAIT_TIME_MS` constant

Reply:
Looks like there is already a public method called `getLatestStreamOffsetAtStartupTime` in `LLRealtimeSegmentDataManager`. This is also used by the offset checker. I think it is the same information that you are looking for.

Reply:
agreed on making it public